prow/apis/prowjobs/v1/types.go

```diff
@@ -239,6 +239,11 @@ type ProwJobStatus struct {
 	// identifier that Jenkins gave to the build for this
 	// ProwJob.
 	JenkinsBuildID string `json:"jenkins_build_id,omitempty"`
+
+	// NeedReport marks if a prowjob needs to be reported upon
+	// finished, so crier won't waste github api tokens to do
+	// unnecessary queries.
+	NeedReport bool `json:"need_report, omitempty"`
```

Comment:
We need to report on a ProwJob at multiple points (when it switches from triggered to pending and when it switches from pending to complete) so it would be better to keep track of the last PJ state that we successfully reported instead of using a bool. If we use a bool we can't distinguish if NeedReport is false because we reported the triggered->pending change or because we reported the pending->complete change.

It is unclear to me if this should be a field in the PJ status or if it should be an annotation.

Reply:
> We need to report on a ProwJob at multiple points

ack 💯  

> It is unclear to me if this should be a field in the PJ status or if it should be an annotation.

I'd like to make it first class, so we don't need to loop annotations and find the key for each prowjob

Reply:
I found there's already `pj.Spec.Report`, I made it `pj.Status.PrevReportState` instead.

Reply:
one issue is we don't have a state for previous created prowjobs, and it will attempt to report all of them at the initial reployment

Reply:
`pj.Spec.Report` is not an indicator of whether a PJ has been reported or not. It is an indicator of whether the PJ reports at all (determined by value of `skip_report`). We can't just get rid of that field, `pj.Status.PrevReportState` should be an addition instead of a replacement.

With respect to handling previously created prowjobs, we could:
- Write and run a script to update existing jobs.
- Temporarily ignore reporting prowjobs created before a configurable time and set that time to be just before we deploy these changes.
- Make Prow start marking the `PrevReportState` field now and wait a while before we merge this PR. (👎thats slow and doesn't let you test things now)

Reply:
I got rid of my previous `pj.Status.NeedReport`, not `pj.Spec.Report` :-)

I think we can deploy this NO-OP crier, and it will start to mark things as reported but not doing anything, and later on we can enable reportlib?

Reply:
Sure, that sounds like a good plan.